.github/workflows/bazel.yml

```diff
@@ -10,7 +10,7 @@ jobs:
   MacOS:
     strategy:
       matrix:
-        xcode_version: ['14.3.1', '15.2', '15.3']
```

Comment:
The 14.3.1 job was failing on Bazel because of diff in fixtures. I'm not able to install that version on my OS. If we're okay with bumping this this should be ready now

Reply:
Yeah that's a fairly old version now, fine to remove from CI